Figure out missing dependencies not documented

Because why not!

npm install failed for me locally:

[fabrice:~/dev/browser-laptop] master ± npm install
npm WARN engine urlutil.js@0.1.1: wanted: {"node":">= 4.0.0"} (current: {"node":"0.12.7","npm":"2.11.3"})
npm WARN prefer global babel@6.3.26 should be installed with -g
 
> tracking-protection@1.0.8 preinstall /home/fabrice/dev/browser-laptop/node_modules/tracking-protection
> npm install hashset-cpp && npm install cppunitlite

-
> hashset-cpp@1.0.13 install /home/fabrice/dev/browser-laptop/node_modules/tracking-protection/node_modules/hashset-cpp
> node-gyp rebuild

make: Entering directory `/home/fabrice/dev/browser-laptop/node_modules/tracking-protection/node_modules/hashset-cpp/build'
  CXX(target) Release/obj.target/hashset-cpp/addon.o
  CXX(target) Release/obj.target/hashset-cpp/HashSetWrap.o
../HashSetWrap.cpp: In static member function ‘static void HashSetWrap::HashSetWrap::Init(v8::Local<v8::Object>)’:
../HashSetWrap.cpp:32:31: error: ‘class v8::Object’ has no member named ‘GetIsolate’
   Isolate* isolate = exports->GetIsolate();

@bbondy I am having the same issue. Installing node-gyp globally didn't fix it for me.

Last few lines of npm install output:

```
  │ ├── oauth-sign@0.8.0 
  │ ├── qs@5.2.0 
  │ ├── stringstream@0.0.5 
  │ ├── tough-cookie@2.2.1 
  │ └── tunnel-agent@0.4.2 
  ├─┬ rimraf@2.5.0 
  │ └─┬ glob@6.0.4 
  │   ├── minimatch@3.0.0 
  │   └── path-is-absolute@1.0.0 
  ├── semver@5.1.0 
  ├─┬ tar@2.2.1 
  │ └── block-stream@0.0.8 
  └─┬ which@1.2.1 
    └─┬ is-absolute@0.1.7 
      └── is-relative@0.1.3 

npm ERR! Darwin 15.2.0
npm ERR! argv "node" "/Users/markshust/Sites/browser-laptop/node_modules/.bin/npm" "install" "bloom-filter-cpp"
npm ERR! node v0.12.7
npm ERR! npm  v3.4.1
npm ERR! code ELIFECYCLE

npm ERR! bloom-filter-cpp@1.0.19 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bloom-filter-cpp@1.0.19 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bloom-filter-cpp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls bloom-filter-cpp
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp/npm-debug.log
npm ERR! Darwin 15.2.0
npm ERR! argv "node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.7
npm ERR! npm  v2.12.1
npm ERR! code ELIFECYCLE

npm ERR! tracking-protection@1.0.8 preinstall: `npm install hashset-cpp && npm install cppunitlite`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tracking-protection@1.0.8 preinstall script 'npm install hashset-cpp && npm install cppunitlite'.
npm ERR! This is most likely a problem with the tracking-protection package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm install hashset-cpp && npm install cppunitlite
npm ERR! You can get their info via:
npm ERR!     npm owner ls tracking-protection
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/markshust/Sites/browser-laptop/npm-debug.log
```

And here's last few lines of npm-debug.log

```
12170 silly gunzTarPerm extractEntry vendor/depot_tools/weekly
12171 silly gunzTarPerm extractEntry vendor/depot_tools/win_toolchain/README.md
12172 silly gunzTarPerm extractEntry vendor/depot_tools/win_toolchain/7z/LICENSE
12173 silly gunzTarPerm extractEntry vendor/depot_tools/win_toolchain/OWNERS
12174 silly gunzTarPerm extractEntry vendor/depot_tools/win_toolchain/get_toolchain_if_necessary.py
12175 silly gunzTarPerm extractEntry vendor/depot_tools/win_toolchain/package_from_installed.py
12176 silly gunzTarPerm extractEntry vendor/depot_tools/wtf
12177 silly gunzTarPerm extractEntry vendor/depot_tools/git-nav-upstream
12178 verbose write writing to /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp/package.json
12179 info preinstall abp-filter-parser-cpp@1.0.22
12180 verbose unsafe-perm in lifecycle true
12181 verbose unlock done using /Users/markshust/.npm/_locks/electron-prebuilt-770dbfa353fee535.lock for /Users/markshust/Sites/browser-laptop/node_modules/electron-prebuilt
12182 info abp-filter-parser-cpp@1.0.22 Failed to exec preinstall script
12183 verbose unlock done using /Users/markshust/.npm/_locks/abp-filter-parser-cpp-874d1ff503bfb65b.lock for /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp
12184 verbose stack Error: tracking-protection@1.0.8 preinstall: `npm install hashset-cpp && npm install cppunitlite`
12184 verbose stack Exit status 1
12184 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
12184 verbose stack     at EventEmitter.emit (events.js:110:17)
12184 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
12184 verbose stack     at ChildProcess.emit (events.js:110:17)
12184 verbose stack     at maybeClose (child_process.js:1015:16)
12184 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
12185 verbose pkgid tracking-protection@1.0.8
12186 verbose cwd /Users/markshust/Sites/browser-laptop
12187 error Darwin 15.2.0
12188 error argv "node" "/usr/local/bin/npm" "install"
12189 error node v0.12.7
12190 error npm  v2.12.1
12191 error code ELIFECYCLE
12192 error tracking-protection@1.0.8 preinstall: `npm install hashset-cpp && npm install cppunitlite`
12192 error Exit status 1
12193 error Failed at the tracking-protection@1.0.8 preinstall script 'npm install hashset-cpp && npm install cppunitlite'.
12193 error This is most likely a problem with the tracking-protection package,
12193 error not with npm itself.
12193 error Tell the author that this fails on your system:
12193 error     npm install hashset-cpp && npm install cppunitlite
12193 error You can get their info via:
12193 error     npm owner ls tracking-protection
12193 error There is likely additional logging output above.
12194 verbose exit [ 1, true ]
12195 verbose unbuild node_modules/tracking-protection
12196 info preuninstall tracking-protection@1.0.8
12197 info uninstall tracking-protection@1.0.8
12198 verbose unbuild rmStuff tracking-protection@1.0.8 from /Users/markshust/Sites/browser-laptop/node_modules
12199 info postuninstall tracking-protection@1.0.8
12200 silly gentlyRm /Users/markshust/Sites/browser-laptop/node_modules/tracking-protection is being purged from base /Users/markshust/Sites/browser-laptop
12201 verbose gentlyRm don't care about contents; nuking /Users/markshust/Sites/browser-laptop/node_modules/tracking-protection
12202 silly vacuum-fs purging /Users/markshust/Sites/browser-laptop/node_modules/tracking-protection
12203 silly vacuum-fs quitting because other entries in /Users/markshust/Sites/browser-laptop/node_modules
12204 verbose unbuild node_modules/abp-filter-parser-cpp
12205 info preuninstall abp-filter-parser-cpp@1.0.22
12206 info uninstall abp-filter-parser-cpp@1.0.22
12207 verbose unbuild rmStuff abp-filter-parser-cpp@1.0.22 from /Users/markshust/Sites/browser-laptop/node_modules
12208 info postuninstall abp-filter-parser-cpp@1.0.22
12209 silly gentlyRm /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp is being purged from base /Users/markshust/Sites/browser-laptop
12210 verbose gentlyRm don't care about contents; nuking /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp
12211 silly vacuum-fs purging /Users/markshust/Sites/browser-laptop/node_modules/abp-filter-parser-cpp
12212 silly vacuum-fs quitting because other entries in /Users/markshust/Sites/browser-laptop/node_modules
```

I'm getting a very similar error in Linux Mint 17 64-bit

```npm ERR! hashset-cpp@1.0.13 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the hashset-cpp@1.0.13 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the hashset-cpp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls hashset-cpp
npm ERR! There is likely additional logging output above.
```

--- a/package.json
+++ b/package.json
@@ -120,5 +120,9 @@
   },
   "pre-commit": [
     "lint"
-  ]
+  ],
+  "engineStrict": true,
+  "engines": {
+    "node": ">= 5.0.0"
+  }
 }

